client/modules/Manage/Bookings/components/BookingAdm.js

```diff
@@ -0,0 +1,365 @@
+    const text = this.state.booking.isDisabled ? disabled : active;
+    
+    const opts = [
+      {state: false, text: active},
+      {state: true, text: disabled},
+    ].map(m => (<DropdownItem key={m.text} onClick={(e) => this.ddOnSelect(field, m.state, e)}>{m.text}</DropdownItem>));
+    
+    return (
+      <FormGroup>
+          {this.renderLabel(field, field)} <br />
+          <ButtonDropdown isOpen={this.state.dropdownsOpen[field]} toggle={(e) => this.toggleDropDown(field, e)}>            
+            <DropdownToggle caret size="sm">{text}</DropdownToggle>
+            <DropdownMenu>
+              {opts}
+            </DropdownMenu>
+          </ButtonDropdown>          
+      </FormGroup>
+    );
+  }
+
+  renderUserDropDown(field, options) {
+    const opts = options.map(m => (
+        <DropdownItem key={m._id} onClick={(e) => this.ddOnSelect(field, m, e)}>{m.email}</DropdownItem>
+      )
+    );
+    const text = this.state.booking[field] ? this.state.booking[field].email : `Select ${this.labels[field]}`;
+    return (
+      <FormGroup>
+          {this.renderLabel(field, field)} <br />
+          <ButtonDropdown isOpen={this.state.dropdownsOpen[field]} toggle={(e) => this.toggleDropDown(field, e)}>            
+            <DropdownToggle caret size="sm">{text}</DropdownToggle>
+            <DropdownMenu>
+              {opts}
+            </DropdownMenu>
+          </ButtonDropdown>          
+      </FormGroup>
+    );
+  }
+
+  renderCarDropDown(field, options) {
+    const opts = options.map(m => (
+        <DropdownItem key={m._id} onClick={(e) => this.ddOnSelect(field, m, e)}>{m.make} {m.model} {m.rego}</DropdownItem>
+      )
+    );
+    const text = this.state.booking[field] ? 
+    this.state.booking[field].make + " " + this.state.booking[field].model + " " + this.state.booking[field].rego : 
+    `Select ${this.labels[field]}`;
+    return (
+      <FormGroup>
+          {this.renderLabel(field, field)} <br />
+          <ButtonDropdown isOpen={this.state.dropdownsOpen[field]} toggle={(e) => this.toggleDropDown(field, e)}>            
+            <DropdownToggle caret size="sm">{text}</DropdownToggle>
+            <DropdownMenu>
+              {opts}
+            </DropdownMenu>
+          </ButtonDropdown>          
+      </FormGroup>
+    );
+  }
+
+  submit(evt) {
+    evt.preventDefault()
+    this.setLoading(true);
+
+    manageSvc.bookings.updateBooking(this.state.booking)
+    .then(result => {
+      this.setLoading(false);
+      this.props.onSaved(this.state.booking);
+    })
+    .catch(e => {
+      this.setLoading(false);
+      this.props.onError();
+      console.log(e)
+    });
+  }
+
+  saveButton(isDisabled) {
+    return isDisabled
+      ? <Button color="primary" disabled>Save</Button> 
+      : <Button color="primary" onClick={(e) => this.submit(e)}>Save</Button>;
+
+  }
+
+  setLoading(isLoading) {
+    this.setState({isLoading: isLoading})
+  }
+
+  handleStartDateChange(date)
```

Comment:
handleStartDateChange and handleEndDateChange could be a single method that takes an additional param for the field